uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/services/leave-popup.service.spec.js

```diff
@@ -0,0 +1,121 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'leave-absences/shared/services/leave-popup.service'
+], function () {
+  'use strict';
+
+  describe('LeavePopup', function () {
+    var $provide, $rootScope, $q, notification, $uibModal, LeavePopup, LeaveRequest, LeaveRequestInstance;
+
+    beforeEach(module('common.angularDate', 'leave-absences.models', 'leave-absences.services', 'leave-absences.settings', function (_$provide_) {
+      $provide = _$provide_;
+    }));
+
+    beforeEach(inject(function () {
+      $uibModal = jasmine.createSpyObj('$uibModal', ['open']);
+      $provide.value('$uibModal', $uibModal);
+    }));
+
+    beforeEach(inject(function (_$rootScope_, _$q_, _notificationService_, _LeavePopup_, _LeaveRequest_, _LeaveRequestInstance_) {
+      $rootScope = _$rootScope_;
+      $q = _$q_;
+      notification = _notificationService_;
+      LeavePopup = _LeavePopup_;
+      LeaveRequest = _LeaveRequest_;
+      LeaveRequestInstance = _LeaveRequestInstance_;
+
+      spyOn(LeaveRequest, 'find');
+      spyOn(notification, 'alert');
+    }));
+
+    describe('openModal()', function () {
+      describe('when called without a Leave request and with a leave type parameter', function () {
+        beforeEach(function () {
+          LeavePopup.openModal(null, 'Leave');
+        });
+
+        it('initializes the leave popup with LeaveRequestController', function () {
+          expect($uibModal.open).toHaveBeenCalledWith(jasmine.objectContaining({
+            controller: 'LeaveRequestCtrl'
+          }));
+        });
+      });
+
+      describe('when called with a Leave request', function () {
+        var requestType = 'SomeValue';
+
+        beforeEach(function () {
+          LeavePopup.openModal({ request_type: requestType });
+        });
+
+        it('initializes the leave popup with capitalizing the request type', function () {
```

Comment:
this description doesn't have much meaning. what you are testing here is that the correct sub controller is initialized